File: theArcade/Intro/exploringWaters/arrayChange.py
def arrayChange(x):
    result = 0
    for i in range(1,len(x)):
        if x[i-1] is x[i]:
            result += 1
            x[i] += 1
        elif x[i-1] > x[i]:
            result += (x[i-1]-x[i]) + 1
            x[i] = x[i-1] + 1
    print(x)
    return result


print("should out put 3") 
print(arrayChange([1,1,1]))

File: theArcade/Intro/4exploringWaters/arrayChange.js
/**
 * You are given an array of integers. 
 * On each move you are allowed to increase exactly one of its element by one. 
 * Find the minimal number of moves required to obtain a strictly 
 * increasing sequence from the input.
 * 
 * Example
 * For inputArray = [1, 1, 1], the output should be
 * arrayChange(inputArray) = 3.
 * @param inputArray : array of integers
 */
function arrayChange(inputArray) {
    let count = 0;
    for (let i=1; i<inputArray.length; i++) {
        let prev = inputArray[i-1];
        let curr = inputArray[i];
        if (prev === curr) {
            count ++;
            inputArray[i] ++;
        } else if (prev > curr) {
            // console.log(`(prev - curr) is ${(prev - curr)}`);
            // console.log(`prev is ${prev}`);
            // console.log(`curr is ${curr}`);
            count += (prev - curr) + 1;
            inputArray[i] = inputArray[i-1]+1;
        }
    }
    // console.log(inputArray);
    return count;
}
const ex = [-100, 0, -2, 0];
console.log(arrayChange(ex));
